### PR TITLE
feat: Allow exclude vectors when loading records

### LINKED ISSUE #2398: rg.load takes too long because of the vector field
**Is your feature request related to a problem? Please describe.**
`rg.load` takes too long because of the `vector` field, even when I don't need it.

**Describe the solution you'd like**
Allow passing a param or through the query
use the elasticsearch `_source` query field to exclude `vector` when not needed.

```
"_source": {        
        "exclude": ["vector"]
    }
```

**Describe alternatives you've considered**
wait

**Additional context**
Low priority

### PR DESCRIPTION
**Pull request by frascuchon**
# NOTE

~~PR #2576 must be merged first to make this PR "mergeable"~~

# Description

Work implemented in https://github.com/argilla-io/argilla/pull/2577 and https://github.com/argilla-io/argilla/pull/2540 tries to introduce a mechanism for:

1. Loading data with field projections (selecting which record fields will be loaded)
2. Allow partial updates when uploading projected records

Some problems of this flow are raised during the work:

- There is a field mismatch between client and API records. The proposed field projection is referring to the Record API data model, and aligning it with client representation can be messy and give an unexpected behavior
- The update flag mechanism is a bit cumbersome because of the client data projection and the record class validation alignments. So, even for a single field update, you need to retrieve extra fields to keep aligned the validations.

The field projection in `rg.load` function can introduce confusion because both record models, client, and API, are not aligned. So, I prefer to remove this functionality. Instead of that, we can introduce some extra parameters to load for setup if users want to exclude `vectors` or `metrics` when loading data. This will help to mitigate problems founds in #2398 

This option, together with changes introduced in #2576, allows users to load partial record info and still update it without losing stored data.

For example, if users want to add some metadata to already stored records, they could try the following flow:

```python
ds = rg.load(name=dataset, exclude_vectors=True)

for record in ds:
   record.metadata = ...

rg.log(name=dataset, records=ds) # This will update records only by adding the metadata info
```

Closes [#2398](https://github.com/argilla-io/argilla/issues/2398)

Refs: https://github.com/argilla-io/argilla/pull/2577 and https://github.com/argilla-io/argilla/pull/2540

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [x] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

Existing tests are been modified including this new behavior

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Review by davidberenstein1957 — APPROVED** [review]
This approach makes sense to me.

File: src/argilla/client/client.py [hunk 1]
```diff
@@ -490,6 +492,13 @@ def load(
                 can be used to load using batches.
             as_pandas: DEPRECATED! To get a pandas DataFrame do
                 ``rg.load('my_dataset').to_pandas()``.
+            batch_size: If provided, load `batch_size` samples per request. A lower batch
+                size may help avoid timeouts.
+            exclude_vectors: When set to `True`, indicates that the record data will be retrieved excluding its vectors,
+                if any. By default, this parameter is set to `False`, meaning that vector data will be included.
+            exclude_metrics: When set to `True`, indicates that the record data will be retrieved excluding its metrics.
+                By default, this parameter is set to `False`, meaning that metrics will be included.
```
Comment (tomaarsen) [inline src/argilla/client/client.py]:
```suggestion
            batch_size: If provided, load `batch_size` samples per request. A lower batch
                size may help avoid timeouts.
            exclude_vectors: When set to `True`, indicates that the record data will be retrieved excluding its vectors,
                if any. By default, this parameter is set to `False`, meaning that vector data will be included.
            exclude_metrics: When set to `True`, indicates that the record data will be retrieved excluding its metrics.
                By default, this parameter is set to `False`, meaning that metrics will be included.
            as_pandas: DEPRECATED! To get a pandas DataFrame do
                ``rg.load('my_dataset').to_pandas()``.
```
Let's stick with the ordering from the method signature.
Edit: Wow, the diff looks like a mess. I've just moved `as_pandas` down to below `exclude_metrics`.

File: src/argilla/client/client.py [hunk 2]
```diff
@@ -646,6 +659,11 @@ def _load_records_internal(
             if sort is not None:
                 _LOGGER.warning("Results are sorted by vector similarity, so 'sort' parameter is ignored.")
 
+            if exclude_metrics or exclude_vectors:
+                _LOGGER.warning(
+                    "Metrics and vectors cannot be excluded when using vector search. Those params will be ignored"
```
Comment (tomaarsen) [inline src/argilla/client/client.py]:
```suggestion
                    "Metrics and vectors cannot be excluded when using vector search. These parameters will be ignored."
```

**Review by tomaarsen — APPROVED** [review]
The `exclude_vectors = False` is a bit of a double negative which I don't love, so I have preference for `include_vectors = True`. That one requires a bit less thought to understand. However, I'm cool with merging this with `exclude_...`, too.

I have a few nits, this is ready when they're tackled as far as I'm concerned.

**Comment by frascuchon** [comment]
> The `exclude_vectors = False` is a bit of a double negative which I don't love, so I have preference for `include_vectors = True`. That one requires a bit less thought to understand. However, I'm cool with merging this with `exclude_...`, too.
> 
> I have a few nits, this is ready when they're tackled as far as I'm concerned.

Yes, makes sense to me also. It's more standard to use `include_...` instead of `exclude...`

**Review by tomaarsen — APPROVED** [review]
I think this is all set. When the tests go green anyways. Thanks for this and well done :)

**Pull request merged**